Cray-HPE/bos
# Release 2.0.0

## Summary and Scope

- Add an error for failure when no other error exists
- Fix the NOT filter for the configuration operator
- Add status operator error handling for missing components
- CASMCMS-8044: Clear component actual state when shut down.
- CASMCMS-8012: Remove configuration from actual state and discovery
- RPM Version and RPM Release support using gitflow
- Change Source file to include release information.
- Use macro injection to correct for project build source
- Use algol60 RPM build image for SLES15SP3.
- Complete release tag information for sed replacing
- Update Jenkinsfile.github.
- Remove build debug information, switch over to new alphanumeric insensitive source import
- Use Artifactory Image
- Cat out bos-reporter.spec
- Pass in release information to setup function.
- Revert to arti build environment
- Fix license headers and changelog for PR
- Use .version instead of .rpm_version
- Update Makefile to use RPM release
- CASMCMS-8064: Fix BOS v2 boot time option name
- CASMCMS-8058: Add sles sp4 build
- CASMCMS-8042: Update BOS Image Building Parameters
- mend
- Move to non-beta horizontal pod autoscaler
- Migrate to new RPM build environments
- Add additional requirements for python3 buildroot
- Add additional requirements for python3 buildroot
- Move over to arti.hpc based build images
- Pin constraints to previous version for flask
- Downrev markupsafe version
- Remove faulty build test
- more build fun
- Remove null list and float types from spec
- Remove null list and float types from spec
- Update Changelog
- CASMCMS-8010: Remove v1 sessiontemplate name length restriction
- CASMCMS-8080: Restore v1 sesstiontemplate templateurl option
- CASMCMS-7924: Update v1/v2 migration to retain a v1 copy of templates
- Fix indentation and formatting of Jenkinsfile.github
- error_summary repromote to object instead of number.
- CASMCMS-8088: Generate valid unstable chart artifacts
- Use appropriate docker version string in Chart
- CASMCMS-8079: Fix session creation description for stage
- CASMCMS-7885: Increase operator resource limits to avoid throttling
- CASMCMS-7972: Update to use internal HPE network
- Use the latest 1.3.X version of BOA
- CASMCMS-8103: Fix incorrect state when rebooting with the same config
- CASMCMS-8085: Fix error loading old v1 session status records
- CASMCMS-8102: Zero error state of components during session creation
- CASMCMS-8102: Disable unused test RPM to get build unstuck.
- Correct indentation.
- Remove building step for test RPM
- CASMCMS-8088: Generate valid unstable chart artifacts
- CASMCMS-8157: Fix applystaged issues for reboots
- CASMCMS-8101: Migration job should talk only to new cray-bos pods
- CASMCMS-8101: Update changelog
- overrides for templates updated Changelog
- CASMCMS-8163: Make v1 and v2 report db_status instead of <db kind>_status
- CASMCMS-8158: Allow users to globally influence lifecycle of staged data
- Update the option to be named 'clear_stage'
- Update terms
- Update openapi.yaml option name.
- Correct import location of library
- Add call into function routine.


## Issues and Related PRs

Merges the release branch into the master branch

## Testing

None, it's just a merge.

### Tested on:

Not test.
### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

None


## Pull Request Checklist
